package/MDAnalysis/analysis/hbond_analysis.py

```diff
@@ -0,0 +1,199 @@
+  - *angle* (degrees): angle of the hydrogen bond
+
+Hydrogen bond data are returned in a :class:`numpy.ndarray` on a "one line, one observation" basis
+and can be accessed via :attr:`HydrogenBondAnalysis.hbonds`::
+
+    results = [
+        [
+            <frame>,
+            <donor index (0-based)>,
+            <hydrogen index (0-based)>,
+            <acceptor index (0-based)>,
+            <distance>,
+            <angle>
+        ],
+        ...
+    ]
+
+Classes
+-------
+
+.. autoclass:: HydrogenBondAnalysis
+   :members:
+
+   .. attribute:: hbonds
+
+"""
+
+import numpy as  np
+
+from . import base
+from MDAnalysis.lib.distances import capped_distance
+from MDAnalysis.lib.distances import calc_angles
+
+class HydrogenBondAnalysis(base.AnalysisBase):
+    """
+    Perform an analysis of hydrogen bonds in a Universe.
+    """
+
+    def __init__(self, universe, donors_sel, hydrogens_sel, acceptors_sel, d_h_cutoff, d_a_cutoff, d_h_a_angle_cutoff):
+        """
+        Set up atom selections and geometric criteria for finding hydrogen bonds in a Universe.
+
+        :param universe : Universe
+            MDAnalysis Universe object
+        :param donors_sel : str
+            Selection string for the hydrogen bond donor atoms
+        :param hydrogens_sel :  str
+            Selection string for the hydrogen bond hydrogen atoms
+        :param acceptors_sel : str
+            Selection string for the hydrogen bond acceptor atoms
+        :param d_h_cutoff : float (optional)
+            Distance cutoff used for finding donor-hydrogen pairs [1.2]
+        :param d_a_cutoff : float (optional)
+            Distance cutoff for hydrogen bonds. This cutoff refers to the D-A distance. [3.0]
+        :param d_h_a_angle_cutoff: float (optional)
+            D-H-A angle cutoff for hydrogen bonds, in degrees. [150]
+
+        """
+
+        self.u = universe
+        self.donors_sel = donors_sel
+        self.hydrogens_sel = hydrogens_sel
+        self.acceptors_sel = acceptors_sel
+        self.d_h_cutoff = d_h_cutoff
+        self.d_a_cutoff = d_a_cutoff
+        self.d_h_a_angle = d_h_a_angle_cutoff
+
+
+    def run(self, start=None, stop=None, step=None):
+        """
+        Find hydrogen bonds present over the interval defined by start:stop:step.
+
+        Stores the results as
+        :attr:`HydrogenBondAnalysis.hbonds`
+
+        :param start : int (optional)
+            Zero-based index of the first frame to be analysed. [``None``]
+        :param stop : int (optional)
+            Zero-based index of the last frame to be analysed (exclusive). [``None``]
+        :param step : int (optional)
+            Use every `step` frame between `start` (included) and `stop` (excluded). [``None``]
+
+        """
+
+        self.hbonds = [[], [], [], [], [], []]
+        for ts in self.u.trajectory[start:stop:step]:
+
+            box = ts.dimensions
+
+            # Find D-H pairs
+            all_donors = self.u.select_atoms(self.donors_sel)
```

Comment:
For static selections you are doing the selection every step. It's probably a small cost compared to the other calculations. (The original HydrogenBondAnalysis class can create updating selections if needed.)

Reply:
That's a good point, I just based this code on the default behaviour of the current class to update selections, but can introduce static selections.